## In this notebook I start a Skyline DIA project, import my protein database and raw data. 
  * Emma executed the final PECAN run for me with my DIA data, and sent me the resulting .blib file. 
  * I used the following references for this Skyline workflow: [tutorial slidedeck for SRM](https://github.com/RobertsLab/project-pacific.oyster-larvae/blob/master/Skyline-example-files-ETS.sky/slides01.pdf), Emma's [suggested settings](https://www.evernote.com/shard/s242/sh/2d6c487c-7d9b-423d-a7a0-9a6c64199302/82b8812164cd461f), and this [DIA Analysis Tutorial on the Roberts Lab site](https://github.com/sr320/LabDocs/wiki/DIA-data-Analyses).

### Start a Skyline project: open SkylineDaily (I used version 3.6.1.11571) 

![New-DIA-Skyline-00](../../images/New-DIA-Skyline-00.png?raw=true)

### Select "Blank Document", and saved. This project is named **Geoduck-DIA-2017.sky** 
### Open Settings -> Peptide Settings and selected the following: 
### Add background proteome, aka protein database, to project. This is the geoduck gonad transcriptome fasta file with the PRTC protein pasted to the end of the file, which is located [on Owl, "GeoTranscriptomePRTC.fasta"](http://owl.fish.washington.edu/generosa/Generosa_DNR/GeoTranscriptomePRTC.fasta). Then work through the rest of the tabs with the following settings: 

![New-DIA-Skyline-01](../../images/New-DIA-Skyline-01.png?raw=true)

![New-DIA-Skyline-02](../../images/New-DIA-Skyline-02.png?raw=true)

![New-DIA-Skyline-03](../../images/New-DIA-Skyline-03.png?raw=true)

![New-DIA-Skyline-04](../../images/New-DIA-Skyline-04.png?raw=true)

![New-DIA-Skyline-05](../../images/New-DIA-Skyline-05.png?raw=true)

### Under the **LIBRARY** tab, opened and selected the .blib file that was generated by Pecan. This file defines the list of proteins that Pecan found to be present in at least one of my geoduck sample files. 

![New-DIA-Skyline-06](../../images/New-DIA-Skyline-06.png?raw=true)

![New-DIA-Skyline-07](../../images/New-DIA-Skyline-07.png?raw=true)

![New-DIA-Skyline-08](../../images/New-DIA-Skyline-09.png?raw=true)

![New-DIA-Skyline-09](../../master/images/New-DIA-Skyline-03.png?raw=true)


### After adjusting all Peptide Settings, hit "OK."  Then, opened **SETTINGS -> TRANSITION SETTINGS** and adjusted all tabs as follows:

![New-DIA-Skyline-10](../../images/New-DIA-Skyline-10.png?raw=true)

![New-DIA-Skyline-11](../../images/New-DIA-Skyline-11.png?raw=true)

![New-DIA-Skyline-12](../../images/New-DIA-Skyline-12.png?raw=true)

![New-DIA-Skyline-13](../../images/New-DIA-Skyline-13.png?raw=true)

![New-DIA-Skyline-14](../../images/New-DIA-Skyline-14.png?raw=true)

### Select "OK". Next step, populate the analyte tree with the protein database.  Import via FILE -> IMPORT -> FASTA and open the GeoTranscriptomePRTC.fasta.

![New-DIA-Skyline-15](../../images/New-DIA-Skyline-15.png?raw=true)

### The fasta loads, tells you how many proteins were imported, then populates the left side of the screen with the protein "names" (protein ID from my fasta):

![New-DIA-Skyline-16](../../images/New-DIA-Skyline-16.png?raw=true)

![New-DIA-Skyline-17](../../images/New-DIA-Skyline-17.png?raw=true)

![New-DIA-Skyline-18](../../images/New-DIA-Skyline-18.png?raw=true)

### Now it's time to import my sample files! Since I demultiplexed my .raw files during the conversion step, I need to import the converted .mzML files into Skyline (as opposed to using .raw files). 

![New-DIA-Skyline-19](../../images/New-DIA-Skyline-19.png?raw=true)

![New-DIA-Skyline-20](../../images/New-DIA-Skyline-20.png?raw=true)

### Watching import progress ... 

![New-DIA-Skyline-21](../../images/New-DIA-Skyline-21.png?raw=true)

![New-DIA-Skyline-22](../../images/New-DIA-Skyline-22.png?raw=true)

### And it's complete. I've rearranged the windo via VIEW -> RETENTION TIME -> REPLICATE COMPARISON, and VIEW -> PEAK AREAS -> REPLICATE COMPARISON. 
![New-DIA-Skyline-23](../../images/New-DIA-Skyline-23.png?raw=true)
